rockcraft/pebble.py

```diff
@@ -101,26 +188,112 @@ class Service(CraftBaseModel):
     """
 
     override: Literal["merge", "replace"]
-    command: str
+    """Control how this service definition is combined with any other pre-existing
+    definition with the same name in the Pebble plan.
+
+    The value ``merge`` will ensure that values in this layer specification are merged
+    over existing definitions, whereas ``replace`` will entirely override an existing
+    service spec in the plan with the same name.
+    """
+    command: str = pydantic.Field(
+        examples=["/usr/bin/ls", "/usr/bin/somedaemon --db=/db/path [ --port 8080 ]"]
+    )
+    """The command to run the service.
+    This command is executed directly, not interpreted by a shell. May be optionally
+    suffixed by default arguments within square brackets, which may be overridden via
+    pebble's ``--args`` parameter.
+    """
     summary: str | None = None
+    """A short summary of the service."""
     description: str | None = None
+    """A detailed, potentially multi-line, description of the service."""
     startup: Literal["enabled", "disabled"] | None = None
+    """Whether the service is enabled automatically when the rock starts.
+
+    If not provided, defaults to ``disabled``."""
     after: list[str] | None = None
+    """The names of other services that this service should start after."""
     before: list[str] | None = None
+    """The names of other services that this service should start before."""
     requires: list[str] | None = None
+    """The names of other services that this service requires in order to start."""
     environment: dict[str, str] | None = None
+    """Environment variables to set for this process."""
     user: str | None = None
+    """Username for starting service as a different user.
+
+    It is an error if the user doesn't exist.
+    """
     user_id: int | None = None
+    """User ID for starting service as a different user.
+    If both user and user-id are specified, the user's UID must match user-id.
+    """
     group: str | None = None
+    """Group name for starting service as a different user.
+    It is an error if the group doesn't exist.
+    """
     group_id: int | None = None
+    """Group ID for starting service as a different user.
+    If both group and group-id are specified, the group's GID must match group-id.
+    """
     working_dir: str | None = None
-    on_success: Literal["restart", "shutdown", "ignore"] | None = None
-    on_failure: Literal["restart", "shutdown", "ignore"] | None = None
-    on_check_failure: dict[str, Literal["restart", "shutdown", "ignore"]] | None = None
-    backoff_delay: str | None = None
-    backoff_factor: float | None = None
+    """Working directory for the service command."""
+    on_success: Literal["restart", "shutdown", "failure-shutdown", "ignore"] | None = (
```

Comment:
This as well as `success-shutdown` below I got from the pebble docs. I'm not entirely sure these are relevant values for rocks, though. LMK if this should be excluded.